lib/src/merge.rs

```diff
@@ -816,6 +1030,45 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_update_from_simplified() {
+        // 1-way merge
+        assert_eq!(
+            c(&[], &[0]).update_from_simplified(&c(&[], &[1])),
+            c(&[], &[1])
+        );
+        // 3-way merge
+        assert_eq!(
+            c(&[0], &[0, 0]).update_from_simplified(&c(&[], &[1])),
+            c(&[0], &[0, 1])
+        );
+        assert_eq!(
+            c(&[0], &[1, 0]).update_from_simplified(&c(&[], &[0])),
+            c(&[0], &[0, 0])
+        );
+        assert_eq!(
+            c(&[0], &[1, 2]).update_from_simplified(&c(&[1], &[2, 3])),
+            c(&[1], &[2, 3])
+        );
+        // 5-way merge
+        assert_eq!(
+            c(&[0, 0], &[0, 0, 0]).update_from_simplified(&c(&[], &[1])),
+            c(&[0, 0], &[0, 0, 1])
+        );
+        assert_eq!(
+            c(&[0, 1], &[0, 0, 0]).update_from_simplified(&c(&[3], &[2, 1])),
+            c(&[0, 3], &[0, 2, 1])
+        );
+        assert_eq!(
+            c(&[1, 0], &[0, 0, 0]).update_from_simplified(&c(&[3], &[2, 1])),
+            c(&[3, 0], &[0, 1, 2])
+        );
+        assert_eq!(
+            c(&[0, 1], &[2, 3, 4]).update_from_simplified(&c(&[1, 2], &[3, 4, 5])),
+            c(&[1, 2], &[3, 4, 5])
+        );
+    }
```

Comment:
Could you add a test for the example that (in my mind) started it all? It would also be the only test that depends on the exact way simplification reordering works:

```rust
assert_eq!(c(&[0,1], &[0,2,2]).simplify(), ...);
assert_eq!(c(&[0,1], &[0,2,2]).update_from_simplified(
  c(&[3], &[4,5]), ...);
```

Reply:
Actually, the test I wrote in the previous comment doesn't quite test what I wanted. 

I'll update this comment with a better test in a bit. It can also go into a separate commit if you are in a hurry.